completions/_abbr

```diff
@@ -10,197 +10,196 @@
-          "(-g)-g[expand everywhere]" \
-          "(-r)-r[expand at the start of the line]" \
-          "(--regular)--regular[expand at the start of the line]" \
-          "(-S)-S[available in this session]" \
-          "(--session)--session[available in all sessions]" \
-          "(--user)--user[available in all sessions]" \
-          "(-U)-U[available in all sessions]"
+  case $state in
+    cmds)
+      local commands; commands=(
+        'a:Add a new abbreviation.'
+        'add:Add a new abbreviation.'
+        'c:Erase all session abbreviations.'
+        'clear-session:Erase all session abbreviations.'
+        'e:Erase an abbreviation.'
+        'erase:Erase an abbreviation.'
+        "x:Output the ABBREVIATION's EXPANSION."
+        "expand:Output the ABBREVIATION's EXPANSION."
+        'export-aliases:Export abbreviations as alias commands.'
+        'g:Add a regular abbreviation, the expansion of which is prefixed with git; and add a global abbreviation, the abbreviation and expansion of which are prefixed with git.'
+        'git:Add a regular abbreviation, the expansion of which is prefixed with git; and add a global abbreviation, the abbreviation and expansion of which are prefixed with git.'
+        'help:Show the manpage.'
+        'import-aliases:Add regular abbreviations for every regular alias in the session, and global abbreviations for every global alias in the session.'
+        'import-fish:Import fish abbr-syntax abbreviations.'
+        'import-git-aliases:Add regular abbreviations for every Git alias in the current session. The EXPANSION is prefixed with git[Space].'
+        'list:List the abbreviations with their expansions.'
+        'l:List the abbreviations only.'
+        'list-abbreviations:List the abbreviations only.'
+        'list-commands:List as commands suitable for export.'
+        'profile:Log profile information for debugging.'
+        'R:Rename an abbreviation.'
+        'rename:Rename an abbreviation.'
+        'version:Show the current version.'
+      )
+      _describe 'abbr commands' commands
+      ret=0
+      ;;
+    args)
+      case $line[1] in
+        a|\
+        add)
+          # [<SCOPE>] [<TYPE>] [--dry-run] [(--quiet | --quieter)] [(-f | --force)] ABBREVIATION=EXPANSION
+          _arguments \
+            '(-S --session -U --user)'{-S,--session}'[available in this session]' \
+            '(-U --user -S --session)'{-U,--user}'[available in all sessions]' \
+            '(-r --regular -g --global)'{-r,--regular}'[expand at the start of the line]' \
+            '(-g --global -r --regular)'{-g,--global}'[expand everywhere]' \
+            '(--dry-run)--dry-run[see what would result, without making any actual changes]' \
+            '(-q --quiet -qq --quieter)'{-q,--quiet}'[silence success output]' \
+            '(-qq --quieter -q --quiet)'{-qq,--quieter}'[silence success output and warnings]' \
+            '(-f --force)'{-f,--force}'[ignore warnings]'
```

Comment:
1. Yeah, no problem.
2. I'm afraid it can't be done, it's the syntax for grouping options.

There's an example for this in the documentation:

```shell
_arguments \ 
    -a -b \ 
  + '(operation)' \ 
    {-c,--compress}'[compress]' \ 
    {-d,--decompress}'[decompress]' \ 
    {-l,--list}'[list]'
```

Reply:
- [x] Use double quotes (090ebc3)
- [x] Alphabetize opts (c6c4b01)

Reply:
Gotcha thanks for the info